kore/src/Kore/Step/Simplification/Defined.hs

```diff
@@ -0,0 +1,28 @@
+{-|
+Copyright   : (c) Runtime Verification, 2020
+License     : NCSA
+
+-}
+module Kore.Step.Simplification.Defined
+    ( simplify
+    ) where
+
+import Prelude.Kore
+
+import Kore.Internal.OrPattern
+    ( OrPattern
+    )
+import qualified Kore.Internal.Pattern as Pattern
+import Kore.Internal.TermLike
+
+-- | Marks the terms of an 'OrPattern' resulted
+-- from simplification as 'Defined'. Does not
+-- do any actual simplification.
+simplify
+    :: InternalVariable variable
+    => Defined (OrPattern variable)
+    -> OrPattern variable
+simplify Defined { getDefined = defined } =
+    Pattern.markSimplified
+    . fmap mkDefined
+    <$> defined
```

Comment:
This should check if the child is already known to be defined, and remove the outer marker if it is. This could happen if we mark something defined, and later simplify the child pattern in a way that we can otherwise infer it is defined.